plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineWtpProject.java

```diff
@@ -118,17 +125,37 @@ public void execute(IProgressMonitor monitor) throws InvocationTargetException,
       throw new InvocationTargetException(ex);
     }
 
-    addAppEngineFacet(newProject, subMonitor.newChild(4));
+    addAppEngineFacet(newProject, subMonitor.newChild(6));
 
     if (config.getUseMaven()) {
-      enableMavenNature(newProject, subMonitor.newChild(2));
+      enableMavenNature(newProject, subMonitor.newChild(4));
+      BuildPath.addMavenLibraries(newProject, config.getAppEngineLibraries(), subMonitor.newChild(5));
     } else {
       addJunit4ToClasspath(newProject, subMonitor.newChild(2));
-    }
+      IJavaProject javaProject = JavaCore.create(newProject);
+      
+      List<Library> libraries = config.getAppEngineLibraries();
+      // todo duplicates some code in CloudLibrariesPage
+      Set<LibraryFile> masterFiles = new HashSet<>();
+      for (Library library : libraries) {
+        if (!library.isResolved()) {
+          library.resolveDependencies();
+        }
+        masterFiles.addAll(library.getLibraryFiles());
+      }
 
-    BuildPath.addLibraries(newProject, config.getAppEngineLibraries(), subMonitor.newChild(2));
+      Library masterLibrary = CloudLibraries.getMasterLibrary();
+      // new project so no existing files in master yet
+      masterLibrary.setLibraryFiles(masterFiles);
+      
+      // todo pointless; refactor so addLibraries takes a Collection<LibraryFile>
+      ArrayList<Library> masterLibraries = new ArrayList<>();
+      masterLibraries.add(masterLibrary);
```

Comment:
Ditto. (Using `Collections.singletonList()` and about the `masterLibraries` name.)

Reply:
fixed by refactoring and simplifying the API